x/mongo/driver/operation.go

```diff
@@ -418,6 +415,10 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 				continue
 			}
 
+			if op.ProcessResponseFn != nil {
```

Comment:
It's not obvious that the placement of this call matters, so let's add a comment to explain that.